android/quest/src/main/java/org/smartregister/fhircore/quest/ui/launcher/GeoWidgetLauncherFragment.kt

```diff
@@ -16,125 +16,243 @@
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        showSetLocationDialog()
+        setOnQuestionnaireSubmissionListener()
+        setLocationFromDbCollector()
+        geoWidgetLauncherViewModel.checkSelectedLocation()
+        Timber.i("GeoWidgetLauncherFragment onViewCreated")
+    }
+
+    private fun buildGeoWidgetFragment() {
+        geoWidgetFragment = GeoWidgetFragment.builder()
+            .setUseGpsOnAddingLocation(false)
+            .setOnAddLocationListener { geoWidgetLocation: GeoWidgetLocation ->
+                if (geoWidgetLocation.position == null) return@setOnAddLocationListener
+                geoWidgetLauncherViewModel.launchQuestionnaireWithParams(
+                    geoWidgetLocation,
+                    activity?.tryUnwrapContext() as Context,
+                    geoWidgetConfiguration.registrationQuestionnaire,
+                )
+            }
+            .setOnCancelAddingLocationListener {
 
-    if (savedInstanceState == null) {
-      addGeoWidgetFragment()
+            }
+            .setOnClickLocationListener { geoWidgetLocation: GeoWidgetLocation ->
+                // todo: open profile
+            }
+            .build()
     }
 
-    setLocationFromDbCollector()
-    setOnQuestionnaireSubmissionListener()
-  }
-  private fun setOnQuestionnaireSubmissionListener() {
-    viewLifecycleOwner.lifecycleScope.launch {
-      repeatOnLifecycle(Lifecycle.State.STARTED) {
-        eventBus.events
-          .getFor(MainNavigationScreen.GeoWidgetLauncher.eventId(geoWidgetConfiguration.id))
-          .onEach { appEvent ->
-            if (appEvent is AppEvent.OnSubmitQuestionnaire) {
-              val extractedResourceIds = appEvent.questionnaireSubmission.extractedResourceIds
-              geoWidgetLauncherViewModel.onQuestionnaireSubmission(extractedResourceIds)
+    private fun setOnQuestionnaireSubmissionListener() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            repeatOnLifecycle(Lifecycle.State.STARTED) {
+                eventBus.events
+                    .getFor(MainNavigationScreen.GeoWidgetLauncher.eventId(geoWidgetConfiguration.id))
+                    .onEach { appEvent ->
+                        if (appEvent is AppEvent.OnSubmitQuestionnaire) {
+                            val extractedResourceIds =
+                                appEvent.questionnaireSubmission.extractedResourceIds
+                            geoWidgetLauncherViewModel.onQuestionnaireSubmission(
+                                extractedResourceIds
+                            )
+                        }
+                    }
+                    .launchIn(lifecycleScope)
             }
-          }
-          .launchIn(lifecycleScope)
-      }
+        }
+    }
+
+    private fun setLocationFromDbCollector() {
+        viewLifecycleOwner.lifecycleScope.launch {
+            delay(1000)
```

Comment:
delay added to render service location into map.

Reply:
can you explain why? Also need to move this number to a constant

Reply:
ah interesting, yea worth looking into a bit more but this seems like a reasonable fix for now. Can you add a comment in the code explaining this, and also move the 1k seconds to a semantically named constant here, like PRE_LOADING_TIMEOUT, or something clearer than that

Reply:
Did you figure out why the flow wasn't collecting any items? Could it be an issue with subscribing to cold/hot flow?

Reply:
I tried some small fixes but no luck 
@ellykits would you have sometime to look into this with me?